src/cli/cli_instance.cpp

```diff
@@ -46,6 +46,7 @@ namespace ot {
 
 namespace Cli {
 
+#ifdef OTDLL
```

Comment:
Is this `#ifdef` needed?  This file should only be compiled in if multiple instance support is being used.

Reply:
Proposed solution seems to be similar to openthread/blob/master/src/core/api/link_raw_api.cpp.
We realize that cli_instance.cpp file is included/excluded from build on makefile level, however it will be difficult to implement it in our build system. With proposed change, source files can be just copied into another solution.